lib/board.rb

```diff
@@ -0,0 +1,43 @@
+require 'pry'
+class Board
+  @cells = []
+  def initialize
+    @cells = [" "," "," "," "," "," "," "," "," "]
+  end
+  def cells
+    @cells
+  end
+  def cells=(array)
+    @cells = array
+  end
+  def reset!
+    @cells = [" "," "," "," "," "," "," "," "," "]
+  end
+  def display
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+  def position(input)
+    @cells[input.to_i-1]
+  end
+  def full?
+    select_if_empty = @cells.select { |cell| cell == " " }
+    select_if_empty == [] ? (true) : (false)
+  end
+  def turn_count
+    select_if_empty = @cells.select { |cell| cell == " " }
+    9-select_if_empty.length  
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```